ae/core/playwright_manager.py

```diff
@@ -251,37 +252,66 @@ async def set_overlay_state_handler(self):
+    
     async def overlay_state_handler(self, is_collapsed: bool):
         page = await self.get_current_page()
         self.ui_manager.update_overlay_state(is_collapsed)
         if not is_collapsed:
             await self.ui_manager.update_overlay_chat_history(page)
 
+    async def show_steps_state_handler(self, show_details: bool):
+        self.ui_manager.update_overlay_show_details(show_details)
 
     async def set_user_response_handler(self):
         context = await self.get_browser_context()
         await context.expose_function('user_response', self.receive_user_response) # type: ignore
 
 
-    async def notify_user(self, message: str):
+    async def notify_user(self, message: str, message_type: str = "step"):
         """
         Notify the user with a message.
 
         Args:
             message (str): The message to notify the user with.
+            message_type (str, optional): The type of message. Defaults to "step", other values are "plan", "action", "answer", "question", "info". 
+            To Do: Convert to Enum.
         """
-        logger.debug(f"Notification: \"{message}\" being sent to the user.")
+
+        safe_message_type = escape_js_message(message_type)
+        if self.ui_manager.overlay_show_details == False:
+            if not (message_type == "plan" or message_type == "question" or message_type == "answer" or message_type == "info"):
+                return
+
+        if self.ui_manager.overlay_show_details == True:
+            if not (message_type == "plan" or message_type == "question" or message_type == "answer" or message_type == "info" or message_type == "step"):
+                return
+            
+        if message_type == "plan":
+            message = beautify_plan_message(message)
+            message = "Plan: \n" + message
+        if message_type == "step":
+            if "confirm" in message.lower():
+                message = "Verify: " + message
+            else:
+                message = "Next step: " + message
+        if message_type == "question":
+            message = "Question: " + message
+
+        if message_type == "answer":
+            message = "Response: " + message
         safe_message = escape_js_message(message)
-        self.ui_manager.new_system_message(safe_message)
+        self.ui_manager.new_system_message(safe_message, message_type)
+
         try:
-            js_code = f"addSystemMessage({safe_message}, false);"
+            js_code = f"addSystemMessage({safe_message}, is_awaiting_user_response=false, message_type={safe_message_type});"
             page = await self.get_current_page()
             await page.evaluate(js_code)
             logger.debug("User notification completed")
         except Exception as e:
+            print(f"Failed to notify user with message \"{message}\". However, most likey this will work itself out after the page loads: {e}")
```

Comment:
We can just change the logger line below it to be logger.error (logger.warn) right? This way we have one print